### PR TITLE
Fix dockerignore and update and fix Docker CI to action v2

### PR DESCRIPTION
**Pull request by louis030195**

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -1,32 +1,38 @@
...
-          repository: prismarinejs/flying-squid/flying-squid
-          path: deploy
-          registry: docker.pkg.github.com
-          dockerfile: deploy/Dockerfile
-          tag_with_ref: true
\ No newline at end of file
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
```
Comment (louis030195) [inline .github/workflows/docker.yml:31]: I think the path (context) was "." and not deploy i.e. when I build locally I do `docker build -t prismarinejs/flying-squid -f deploy/Dockerfile .` (from root folder). I updated to the v2 action too which is maybe a little bit less verbose

see example [here](https://github.com/louis030195/niwrad/blob/0c3f348bcf31457d90d08c39b3b35a3c64940f7c/.github/workflows/main.yml#L38)

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -1,32 +1,38 @@
 name: Publish Docker image
 on:
-  release:
-    types: [published]
+  # release:
+  #   types: [published]
+  push:
```
Comment (rom1504) [inline .github/workflows/docker.yml:5]: add pull_request trigger like https://github.com/PrismarineJS/flying-squid/blob/master/.github/workflows/ci.yml#L6 to be able to test

**Comment by louis030195** [comment]
I doubt a random person can use a repository secrets by just sending a PR

**Comment by rom1504** [comment]
https://github.community/t/feature-request-allow-secrets-in-approved-external-pull-requests/18071 yes indeed
merging

**Review by rom1504 — APPROVED** [review]

**Comment by louis030195** [comment]
Well dockerhub seems to login at least, idk why github registry fail, could remove github registry, who care ?

**Comment by rom1504** [comment]
I fixed that but the DockerFile is wrong.
any reason to run npm prepare instead of npm install and why this copy of package.json ?
https://github.com/PrismarineJS/flying-squid/blob/master/deploy/Dockerfile

**Comment by rom1504** [comment]
tried to fix, didn't work
could you create a workflow that will build the docker and publish only if it's not a PR ?
that way you can test without merging in master

**Comment by louis030195** [comment]
official node docker recommendation

```dockerfile
FROM node:12

# Create app directory
WORKDIR /usr/src/app

# Install app dependencies
# A wildcard is used to ensure both package.json AND package-lock.json are copied
# where available (npm@5+)
COPY package*.json ./

RUN npm install
# If you are building your code for production
# RUN npm ci --only=production

# Bundle app source
COPY . .

EXPOSE 8080
CMD [ "node", "server.js" ]
```

I had issue with "requireindex.js" and importing stuff when using npm install but could finally make it work with npm run prepare.

For the workflow "only if its not a PR" you mean

```yaml
  push:
    branches: [ some_branch ]
```
?

**Comment by louis030195** [comment]
npm run install doesn't exist btw

**Comment by louis030195** [comment]
maybe I should just try to push on my own registry on my fork to try

**Comment by rom1504** [comment]
I mean the docker build should run before any docker login. And the docker
login shouldn't be needed in pull request, only in commits in master.
That way it's possible to validate the docker builds in all pull requests

**Comment by rom1504** [comment]
I seem to have fixed it
it's there https://hub.docker.com/repository/docker/prismarinejs/flying-squid

**Comment by rom1504** [comment]
there too https://github.com/orgs/PrismarineJS/packages/container/package/flying-squid%2Fflying-squid

**Comment by louis030195** [comment]
nice ! build works but the container is broken, why did you put npm install instead of npm run prepare which worked no ?

> louis@louis-pc:~$ docker run prismarinejs/flying-squid
> internal/modules/cjs/loader.js:883
>   throw err;
>   ^
> 
> Error: Cannot find module 'flying-squid'
> Require stack:
> - /app/src/lib/plugins/behavior.js
> - /app/src/lib/requireindex.js
> - /app/src/index.js
> - /app/index.js
> - /app/app.js
>     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
>     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
>     at Module.require (internal/modules/cjs/loader.js:952:19)
>     at require (internal/modules/cjs/helpers.js:88:18)
>     at Object.<anonymous> (/app/src/lib/plugins/behavior.js:1:18)
>     at Module._compile (internal/modules/cjs/loader.js:1063:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
>     at Module.load (internal/modules/cjs/loader.js:928:32)
>     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
>     at Module.require (internal/modules/cjs/loader.js:952:19) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [
>     '/app/src/lib/plugins/behavior.js',
>     '/app/src/lib/requireindex.js',
>     '/app/src/index.js',
>     '/app/index.js',
>     '/app/app.js'
>   ]
> }
> 
> 

Weird error though, prob related to some specific config of flying-squid

**Comment by rom1504** [comment]
@louis030195 can you try again now ?

**Comment by louis030195** [comment]
@rom1504 works fine !

**Comment by louis030195** [comment]
> louis@louis-pc:~$ docker run prismarinejs/flying-squid
> October 25th 2020, 11:38:15 [Warning] onItemPlace handler was registered twice for redstone_torch
> October 25th 2020, 11:38:15 [Warning] onItemPlace handler was registered twice for repeater
> October 25th 2020, 11:38:15 [Warning] onItemPlace handler was registered twice for oak_sign
> October 25th 2020, 11:38:15 [Warning] onItemPlace handler was registered twice for spruce_sign
> October 25th 2020, 11:38:15 [Warning] onItemPlace handler was registered twice for birch_sign
> October 25th 2020, 11:38:15 [Warning] onItemPlace handler was registered twice for acacia_sign
> October 25th 2020, 11:38:15 [Warning] onItemPlace handler was registered twice for jungle_sign
> October 25th 2020, 11:38:15 [Warning] onItemPlace handler was registered twice for dark_oak_sign
> October 25th 2020, 11:38:15 [INFO]: Server listening on port 25565
> October 25th 2020, 11:38:15 seed: 840196346
> >

**Pull request merged**